bot/exts/moderation/incidents.py

```diff
@@ -402,10 +515,106 @@ async def on_raw_reaction_add(self, payload: discord.RawReactionActionEvent) ->
 
     @Cog.listener()
     async def on_message(self, message: discord.Message) -> None:
-        """Pass `message` to `add_signals` if and only if it satisfies `is_incident`."""
+        """
+        Pass `message` to `add_signals` and `extract_message_links` if it satisfies `is_incident`.
+
+        If the message (`message`) is an incident report, then run it through `extract_message_links`
+        to get all the message link embeds (embeds which contain information about that particular
+        link).These message link embeds are then sent into the channel.
+
+        Also passes the message into `add_signals` if the message is an incident.
+        """
         if is_incident(message):
             await add_signals(message)
 
+            webhook_embed_list = await extract_message_links(message, self.bot)
+            if webhook_embed_list:
+                await self.send_message_link_embeds(webhook_embed_list, message, self.incidents_webhook)
+
+    @Cog.listener()
+    async def on_message_edit(self, msg_before: discord.Message, msg_after: discord.Message) -> None:
+        """
+        Pass `msg_after` to `extract_message_links` and edit `msg_before` webhook msg.
+
+        Deletes cache value (`message_link_embeds_cache`) of `msg_before` if it exists and removes the
+        webhook message for that particular link from the channel.
+
+        If the message edit (`msg_after`) is a incident then run it through `extract_message_links`
+        to get all the message link embeds (embeds which contain information about that particular
+        link), this message link embeds are then sent into the channel.
+
+        The edited message is also passed into `add_signals` if it is a incident message.
+        """
```

Comment:
Have you tested whether this listener works for messages that aren't in the bot's cache? The reason why my implementation uses a raw listener is that the regular reaction listener will not fire when it cannot give you the message instance, i.e. if the bot restarts and a reaction is placed on an older message. The bot should obviously still react to such events. I believe this is explained in the docstring.

Having a quick look at the docs, it seems that the [edit](https://discordpy.readthedocs.io/en/stable/api.html#discord.on_message_edit) and [delete](https://discordpy.readthedocs.io/en/stable/api.html#discord.on_message_delete) listeners probably behave the same way. In practice, this would mean that if the bot restarts, it will not clear the quotation message for incidents sent before the current session, nor will it react to edits.

Please consult the docs, there are raw counterparts for both listeners.

Reply:
Resolved in 0b35f2a and 0c5561d.